temperature/router.py

```diff
@@ -0,0 +1,84 @@
+import asyncio
+import os
+from datetime import datetime
+
+import httpx
+from dotenv import load_dotenv
+from fastapi import APIRouter, HTTPException
+
+from city.crud import get_all_cities
+from dependencies import DBSessionDep, PaginationDep
+from temperature import crud, schemas
+
+
+load_dotenv()
+API_KEY = os.getenv("WEATHER_API_KEY")
+BASE_URL = "http://api.weatherapi.com/v1/current.json"
+
+router = APIRouter()
+
+
+@router.get("/temperatures/", response_model=list[schemas.Temperature])
+def read_temperatures(
+    db: DBSessionDep, paginate: PaginationDep, city_id: int = None
+):
+    return crud.get_temperature_list(db=db, city_id=city_id, **paginate)
+
+
+@router.get(
+    "/temperatures/{temperature_id}", response_model=schemas.Temperature
+)
+def read_temperature(temperature_id: int, db: DBSessionDep):
+    temperature = crud.get_temperature(db=db, temperature_id=temperature_id)
+
+    if not temperature:
+        raise HTTPException(
+            status_code=404, detail="Temperature record not found"
+        )
+
+    return temperature
+
+
+async def fetch_weather(city: str) -> dict:
+    if not API_KEY:
+        raise ValueError("API_KEY not found in environment variables.")
```

Comment:
The `fetch_weather` function raises a `ValueError` if the API key is not found. Consider using a more descriptive error message or handling this case earlier in the application setup.